ssl/quic/quic_impl.c

```diff
@@ -101,6 +104,26 @@ static ossl_inline int expect_quic_conn(const QUIC_CONNECTION *qc)
 
 }
 
+/*
+ * Ensures that the channel mutex is held for a method which touches channel
+ * state.
+ *
+ * Precondition: Channel mutex is not held (unchecked)
+ */
+static int quic_lock(QUIC_CONNECTION *qc)
+{
+    ossl_crypto_mutex_lock(qc->mutex);
```

Comment:
Um. I'm confused. `qc->mutex` is declared as `CRYPTO_RWLOCK`, but `ossl_crypto_mutex_lock` expects `CRYPTO_MUTEX`. What am I missing?